starknet_devnet/blocks.py

```diff
@@ -292,17 +296,36 @@ async def store_pending(
                 new_root=self.__pending_state_update.new_root,
                 state_diff=self.__pending_state_update.state_diff,
             )
-        self.__state_updates[block_number] = self.__pending_state_update
+        self.__state_updates[block_hash] = self.__pending_state_update
         self.__pending_state_update = None
 
         block = StarknetBlock.load(block_dict)
-        self.__num2block[block_number] = block
-        self.__state_archive.store(block_number, state)
+        self.__hash2block[block.block_hash] = block
+        self.__state_archive.store(block_hash, state)
 
         self.__pending_block = None
         self.__pending_signatures = None
         return block
 
-    def get_state(self, block_number: int) -> StarknetState:
+    def get_state(self, block_hash: int) -> StarknetState:
         """Return state at block with `number`"""
-        return self.__state_archive.get(block_number)
+        return self.__state_archive.get(block_hash)
+
+    async def abort_block_by_hash(self, block_hash: str) -> str:
+        """
+        Abort block by given block hash.
+        """
+        numeric_hash = _parse_block_hash(block_hash)
+
+        if numeric_hash in self.__hash2block:
```

Comment:
You're right, the return value is used. But still, the function sometimes returns None. Perhaps it could raise an erroor instead? Also, now it seems to be related to https://github.com/0xSpaceShard/starknet-devnet/pull/421#discussion_r1177554595

Reply:
Ahhh, we depend on it returning None to perform a check. In that case, the return type of the function should be `Optional[None]`

Reply:
And we should explicitly return None. I really don't like the implicit return of None

Reply:
Fixed in 98ba19b58fddda3bcc7c9ab5aebfa704643e7fd7